### PR TITLE
Fix devfile V1 update flow when devworkspace is enabled

### PR DESCRIPTION
**Pull request by olexii4**
Signed-off-by: Oleksii Orel <oorel@redhat.com>

### What does this PR do?
Fix devfile V1 update flow when devworkspace is enabled.

### What issues does this PR fix or reference?
fixes https://github.com/eclipse/che/issues/19922

### CODE REVIEW

**Review by akurinnoy — APPROVED** [review]

File: src/store/Workspaces/index.ts [hunk 1]
```diff
@@ -213,15 +219,13 @@ export const actionCreators: ActionCreators = {
...
     try {
-      const state = getState();
-      const cheDevworkspaceEnabled = state.workspacesSettings.settings['che.devworkspaces.enabled'] === 'true';
-      if (cheDevworkspaceEnabled) {
-        await dispatch(DevWorkspacesStore.actionCreators.updateWorkspace(workspace.ref as IDevWorkspace));
-      } else {
+      if (isWorkspaceV1(workspace.ref)) {
         await dispatch(CheWorkspacesStore.actionCreators.updateWorkspace(workspace.ref as che.Workspace));
+      } else {
+        await dispatch(DevWorkspacesStore.actionCreators.updateWorkspace(workspace.ref as IDevWorkspace));
```
Comment (sleshchenko) [inline src/store/Workspaces/index.ts:228]: Should some unit test be updated?

**Review by sleshchenko — APPROVED** [review]

**Comment by sleshchenko** [comment]
@olexii4 after merging please cherry-pick into 7.32.x as well. Thx

**Review by JPinkney — APPROVED** [review]

File: src/store/Workspaces/index.ts (hunk 1 above)
Comment (olexii4) [reply src/store/Workspaces/index.ts:228]: No. We don't have test coverage for Redux store yet.

**Comment by sleshchenko** [comment]
[dashboard-ci-test]

**Comment by olexii4** [comment]
[dashboard-ci-test]

**Pull request merged**